### PR TITLE
Added data structures

### PR DESCRIPTION
**Pull request by vishwanath1306**
Added basic struct for blocks, give comments on what should be added to this.

### CODE REVIEW

File: src/data_structures.rs [hunk 1]
```diff
@@ -9,6 +9,19 @@ impl CacheKey {
     }
 }
 
+pub struct Stats{
+    pub access_count: i64,
+}
+
+pub struct Segment{
```
Comment (aj-austin) [inline src/data_structures.rs:27]: Might also need other metadata like start virtual address, size?

**Review by aj-austin — COMMENTED** [review]
Overall looks good. We might add more fields to the structs as needed

File: src/data_structures.rs (hunk 1 above)
Comment (deeptir18) [reply src/data_structures.rs:27]: Do we want to keep state inside Segment like Option<u64> to represent the lkey, as well as physical page mappings if pinned (for Intel NICs?) e.g. Hashmap<u64, u64> or something?

We also probably definitely want page size (4kb, 2mb or 1gb)

File: src/data_structures.rs [hunk 2]
```diff
@@ -9,6 +9,19 @@ impl CacheKey {
...
+    pub segment_id: i64,
+    pub segment_stats: Stats
+}
+
+pub struct HotSet{
+    pub current_hotset: Vec<Segment>
```
Comment (deeptir18) [inline src/data_structures.rs:47]: I guess we can either keep track of Hotset as Vec<Segment> or Vec<segment_id> and have a map storing segment_id -> segment somewhere? But this seems fine for now.

File: src/data_structures.rs [hunk 3]
```diff
@@ -9,6 +9,19 @@ impl CacheKey {
     }
 }
 
+pub struct Stats{
+    pub access_count: i64,
```
Comment (deeptir18) [inline src/data_structures.rs:17]: this is a good start. but does it make sense to keep track of how "skewed" the accesses are? Like if all the accesses come from a single page somehow?

File: src/data_structures.rs (hunk 1 above)
Comment (vishwanath1306) [reply src/data_structures.rs:27]: Re Segment, yeah we can put a hashmap inside it to store the mapping. 

Will add page_size

File: src/data_structures.rs (hunk 3 above)
Comment (vishwanath1306) [reply src/data_structures.rs:17]: I think that would be an aggregate stat, like a function on the segment method that can be used for hotset

**Pull request merged**